gdpr/vendorlist-fetching_test.go

```diff
@@ -7,233 +7,697 @@ import (
+				enableTCF1Fallback: false,
+				vendorListVersion:  2,
+			},
+			expected: testExpected{
+				errorMessage: "gdpr vendor list version 2 does not exist, or has not been loaded yet. Try again in a few minutes",
+			},
+		},
+	}
+
+	for _, test := range testCases {
+		runTest(t, test, tcf1Version, server)
+	}
+}
+
+func TestTCF2FetcherInitialLoad(t *testing.T) {
+	// Loads two vendor lists during initialization by setting the latest vendor list version to 2.
+	// Ensures TCF1 fetch settings have no effect on TCF2.
+
+	server := httptest.NewServer(http.HandlerFunc(mockServer(serverSettings{
+		vendorListLatestVersion: 2,
+		vendorLists: map[int]string{
+			1: tcf2VendorList1,
+			2: tcf2VendorList2,
+		},
 	})))
 	defer server.Close()
 
-	fetcher := newVendorListFetcher(context.Background(), testConfig(), server.Client(), testURLMaker(server), 1)
-	list, err := fetcher(context.Background(), 2)
-	assertNilErr(t, err)
+	testCases := []test{
+		{
+			description: "Fetch - No Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: false,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
+		},
+		{
+			description: "Fetch - No Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: false,
+				vendorListVersion:  2,
+			},
+			expected: vendorList2Expected,
+		},
+		{
+			description: "Fetch - Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: true,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
+		},
+		{
+			description: "Fetch - Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: true,
+				vendorListVersion:  2,
+			},
+			expected: vendorList2Expected,
+		},
+		{
+			description: "No Fetch - Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: true,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
+		},
+		{
+			description: "No Fetch - Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: true,
+				vendorListVersion:  2,
+			},
+			expected: vendorList2Expected,
+		},
+		{
+			description: "No Fetch - No Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: false,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
+		},
+		{
+			description: "No Fetch - No Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: false,
+				vendorListVersion:  2,
+			},
+			expected: vendorList2Expected,
+		},
+	}
 
-	vendor := list.Vendor(3)
-	assertBoolsEqual(t, true, vendor.Purpose(1))
-	assertBoolsEqual(t, false, vendor.Purpose(2))
+	for _, test := range testCases {
+		runTest(t, test, tcf2Version, server)
+	}
 }
 
-func TestInitialTimeout(t *testing.T) {
```

Comment:
Are we testing timeouts still?

Reply:
Yes. This isn't actually testing the timeout, but rather the wait for the timeout. See `TestTCF1FetcherThrottling` and `TestTCF2FetcherThrottling`